CIScripts/BuildFunctions.ps1

```diff
@@ -0,0 +1,151 @@
+           [Parameter(Mandatory = $true)] [string] $CertPath,
+           [Parameter(Mandatory = $true)] [string] $CertPasswordFilePath,
+           [Parameter(Mandatory = $true)] [string] $MSIPath)
+    
+    $cerp = Get-Content $CertPasswordFilePath
+    & $SigntoolPath sign /f $CertPath /p $cerp $MSIPath
+    if ($LASTEXITCODE -ne 0) {
+    throw "Signing $MSIPath failed"
+}
+}
+
+function Build-DockerDriver {
+    Param ([Parameter(Mandatory = $true)] [string] $DriverSrcPath,
+           [Parameter(Mandatory = $true)] [string] $SigntoolPath,
+           [Parameter(Mandatory = $true)] [string] $CertPath,
+           [Parameter(Mandatory = $true)] [string] $CertPasswordFilePath)
+    
+    $Env:GOPATH=pwd
+    
+    New-Item -ItemType Directory ./bin
+    Push-Location bin
+
+    Write-Host "Installing test runner"
+    go get -u -v github.com/onsi/ginkgo/ginkgo
+
+    Write-Host "Building driver"
+    go build -v $DriverSrcPath
+
+    $srcPath = "$Env:GOPATH/src/$DriverSrcPath"
+    Write-Host $srcPath
+
+    Write-Host "Precompiling tests"
+    $modules = @("driver", "controller", "hns", "hnsManager")
+    $modules.ForEach({
+        .\ginkgo.exe build $srcPath/$_
+        Move-Item $srcPath/$_/$_.test ./
+    })
+
+    Write-Host "Copying Agent API python script"
+    Copy-Item $srcPath/scripts/agent_api.py ./
+
+    Write-Host "Intalling MSI builder"
+    go get -u -v github.com/mh-cbon/go-msi
+
+    Write-Host "Building MSI"
+    Push-Location $srcPath
+    & "$Env:GOPATH/bin/go-msi" make --msi installer.msi --arch x64 --version 0.1 --src template --out $pwd/gomsi
+    Pop-Location
+
+    Move-Item $srcPath/installer.msi ./
+    
+    Sign-MSI -SigntoolPath $SigntoolPath -CertPath $CertPath -CertPasswordFilePath $CertPasswordFilePath -MSIPath "installer.msi"
+
+    Pop-Location
+}
+
+function Build-Extension {
+    Param ([Parameter(Mandatory = $true)] [string] $ThirdPartyCache,
+           [Parameter(Mandatory = $true)] [string] $SigntoolPath,
+           [Parameter(Mandatory = $true)] [string] $CertPath,
+           [Parameter(Mandatory = $true)] [string] $CertPasswordFilePath)
+    
+    Write-Host "Copying Extension dependencies"
+    Copy-Item -Recurse "$ThirdPartyCache\extension\*" third_party\
+    Copy-Item -Recurse third_party\cmocka vrouter\test\
+
+    
+    Write-Host "Building Extension and Utils"
+    scons vrouter
+    if ($LASTEXITCODE -ne 0) {
+    throw "Building vRouter solution failed"
+}
+
+    $vRouterMSI = "build\debug\vrouter\extension\vRouter.msi"
+    $utilsMSI = "build\debug\vrouter\utils\utils.msi"
+    
+    Write-Host "Signing utilsMSI"
+    Sign-MSI -SigntoolPath $SigntoolPath -CertPath $CertPath -CertPasswordFilePath $CertPasswordFilePath -MSIPath $utilsMSI
+    
+    Write-Host "Signing vRouterMSI"
+    Sign-MSI -SigntoolPath $SigntoolPath -CertPath $CertPath -CertPasswordFilePath $CertPasswordFilePath -MSIPath $vRouterMSI
+}
+
+function Build-Agent {
```

Comment:
Please use approved words in function names (https://msdn.microsoft.com/en-us/library/ms714428(v=vs.85).aspx)